src/Imath/ImathVec.h

```diff
@@ -74,9 +75,36 @@ template <class T> class Vec2
     /// Construct from Vec2 of another base type
     template <class S> IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Vec2 (const Vec2<S>& v) noexcept;
 
+    /// Interoperability constructor from another type that behaves as if it
+    /// were an equivalent vector.
+    template<typename V, IMATH_ENABLE_IF(has_xy<V,T>::value)>
+    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Vec2 (const V& v) noexcept
+        : Vec2(T(v.x), T(v.y)) { }
+
+    template<typename V, IMATH_ENABLE_IF(has_subscript<V,T,2>::value
+                                         && !has_xy<V,T>::value)>
+    IMATH_HOSTDEVICE Vec2 (const V& v) : Vec2(T(v[0]), T(v[1])) { }
```

Comment:
Can this be constexpr?

Reply:
I think this one cannot, because the foreign vector might have its operator[] not be constexpr.